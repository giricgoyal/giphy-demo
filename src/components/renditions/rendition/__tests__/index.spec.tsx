import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Rendition, { Props } from '../index'

describe('<Rendition />', () => {
    let container

    const defaultProps: Props = {
        data: {},
    }

    describe('When data is {}', () => {
        beforeEach(() => {
            ;({ container } = render(<Rendition {...defaultProps} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When url is sent in data', () => {
        beforeEach(() => {
            const props = {
                data: {
                    url: 'https://www.batman.com/portrait.gif',
                },
            }
            ;({ container } = render(<Rendition {...props} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When url is not sent in data', () => {
        beforeEach(() => {
            const props = {
                data: {
                    webp: 'https://www.batman.com/portrait.webp',
                    mp4: 'https://www.batman.com/portrait.mp4',
                },
            }
            ;({ container } = render(<Rendition {...props} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When neither url or webp is sent in data', () => {
        beforeEach(() => {
            const props = {
                data: {
                    mp4: 'https://www.batman.com/portrait.mp4',
                },
            }
            ;({ container } = render(<Rendition {...props} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })
})
