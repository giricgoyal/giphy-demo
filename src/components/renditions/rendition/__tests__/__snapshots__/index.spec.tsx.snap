// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rendition /> When data is {} should match snapshot 1`] = `<div />`;

exports[`<Rendition /> When neither url or webp is sent in data should match snapshot 1`] = `
<div>
  <video
    autoplay=""
    loop=""
    src="https://www.batman.com/portrait.mp4"
  />
</div>
`;

exports[`<Rendition /> When url is not sent in data should match snapshot 1`] = `
<div>
  <img
    src="https://www.batman.com/portrait.webp"
  />
</div>
`;

exports[`<Rendition /> When url is sent in data should match snapshot 1`] = `
<div>
  <img
    src="https://www.batman.com/portrait.gif"
  />
</div>
`;
